src/DynamicRupture/FrictionLaws/GpuImpl/GpuBaseFrictionLaw.cpp

```diff
@@ -45,17 +45,24 @@ void GpuBaseFrictionLaw::allocateAuxiliaryMemory() {
   }
 
   {
-    const size_t requiredNumBytes = CONVERGENCE_ORDER * sizeof(real);
-    devDeltaT = static_cast<real*>(sycl::malloc_shared(requiredNumBytes, queue));
+    const size_t requiredNumBytes = misc::numPaddedPoints * sizeof(real);
+    devSpaceWeights = static_cast<real*>(sycl::malloc_shared(requiredNumBytes, queue));
   }
 }
 
 void GpuBaseFrictionLaw::copyStaticDataToDevice() {
-  constexpr auto dim0 = misc::dimSize<init::resample, 0>();
-  constexpr auto dim1 = misc::dimSize<init::resample, 1>();
-  const size_t requiredNumBytes = dim0 * dim1 * sizeof(real);
+  {
```

Comment:
can you explain what does this formating change mean?
Also why you call in one case queue.memcpy and the other this->queue.memcpy ?

Reply:
a new scope allows me to declare `const size_t requiredNumBytes` again. It was a request from Lukas to use `const` as much as possible. In short, it is supposed to make the code clearer.

> Also why you call in one case queue.memcpy and the other this->queue.memcpy ?

adjusted to `queue.memcpy` in this file.

In header files (e.g., `GpuFrictionSolver.h`) sometimes we need to use `this->queue.memcpy` be cause of the naming resolution

Reply:
ok, I see